packages/compiler/core/semantic-walker.ts

```diff
@@ -241,17 +240,6 @@ function navigateType(
   }
 }
 
-/**
- * Resolve if the model is a template type(Non initialized template type).
- */
-export function isTemplate(model: ModelType): boolean {
```

Comment:
Missed that we made a breaking change here, but it shows up on flow to cadl-azure. The new isTemplate checks for declaration or instance.

I'd be tempted to change the new isTemplate to isTemplateDeclarationOrInstance so that we can't get confused. To me at least, a template instance is not really a template.